vulnerability-analyzer/src/test/java/org/hyades/processor/scanner/snyk/SnykProcessorTest.java

```diff
@@ -160,6 +160,33 @@ void testMultipleResultsWithOneMatchingPurlInRequest() throws Exception {
         );
     }
 
+    @Test
+    void testResultsWithValidAndInvalidPurlsInRequest() throws Exception {
+        when(snykClientMock.getIssues(anyString(), anyString(), any(ReportRequest.class)))
+                .thenReturn(getSnykResponseTestData("200-with-error-response.json"));
+
+        final TestRecord<String, ScanTask> inputRecordValid = createTestRecord(TEST_PURL_JACKSON_BIND);
+        final TestRecord<String, ScanTask> inputRecordInvalid = createTestRecord("pkg:deb/debian/curl@7.50.3");
+        inputTopic.pipeInput(inputRecordValid);
+        // sending invalid purl along with valid purl won't return 400 but data for valid one
+        inputTopic.pipeInput(inputRecordInvalid);
+
+        testDriver.advanceWallClockTime(Duration.ofSeconds(6));
+
+        assertThat(outputTopic.getQueueSize()).isEqualTo(2);
+        TestRecord<ScanKey, ScannerResult> outputRecord = outputTopic.readRecord();
+        assertThat(outputRecord.key()).isEqualTo(inputRecordValid.getValue().getKey());
+        assertThat(outputRecord.getValue().getStatus()).isEqualTo(ScanStatus.SCAN_STATUS_SUCCESSFUL);
+        assertThat(outputRecord.getValue().hasFailureReason()).isFalse();
+        assertThat(outputRecord.getValue().getScanner()).isEqualTo(Scanner.SCANNER_SNYK);
+        assertThat(outputRecord.getValue().getBom().getVulnerabilitiesList()).satisfiesExactlyInAnyOrder(
+                vulnerability -> {
+                    assertThat(vulnerability.getId()).isEqualTo("SNYK-JAVA-COMFASTERXMLJACKSONCORE-3038426");
+                    assertThat(vulnerability.getSource().getName()).isEqualTo("SNYK");
+                }
+        );
```

Comment:
There should be an assertion here that verifies that we reported `SCAN_STATUS_FAILED` for the invalid PURL record.